### PR TITLE
Simplified prow job definitions, changed to use home-brewed gotests image increased lint timeout

### PR DESCRIPTION
**Pull request by radoslawc**

### CODE REVIEW

**Review by gvbalaji — APPROVED** [review]
approve

File: .prow.yaml [hunk 1]
```diff
@@ -1,33 +1,31 @@
 presubmits:
   - name: presubmit-nephio-go-test
     decorate: true
-    run_if_changed: "(\\.go|^Makefile)$"
+    run_if_changed: "^.*.go$"
     spec:
       containers:
-      - image: golang:1.20.2
+      - image: nephio/gotests:2
```
Comment (vjayaramrh) [inline .prow.yaml:7]: curious as to what the gotests image adds that is not already in the official golang:1.20.2 image? Thanks

**Pull request merged**